                                      "Campus Placement Data Cleaning and Analysis using Pandas"

In [1]:
import pandas as pd

In [3]:
df=pd.read_csv(r"C:\Users\muthu\Downloads\Placement_Data_Full_Class.csv")

In [5]:
df

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,1,M,67.00,Others,91.00,Others,Commerce,58.00,Sci&Tech,No,55.0,Mkt&HR,58.80,Placed,270000.0
1,2,M,79.33,Central,78.33,Others,Science,77.48,Sci&Tech,Yes,86.5,Mkt&Fin,66.28,Placed,200000.0
2,3,M,65.00,Central,68.00,Central,Arts,64.00,Comm&Mgmt,No,75.0,Mkt&Fin,57.80,Placed,250000.0
3,4,M,56.00,Central,52.00,Central,Science,52.00,Sci&Tech,No,66.0,Mkt&HR,59.43,Not Placed,NaN
4,5,M,85.80,Central,73.60,Central,Commerce,73.30,Comm&Mgmt,No,96.8,Mkt&Fin,55.50,Placed,425000.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
210,211,M,80.60,Others,82.00,Others,Commerce,77.60,Comm&Mgmt,No,91.0,Mkt&Fin,74.49,Placed,400000.0
211,212,M,58.00,Others,60.00,Others,Science,72.00,Sci&Tech,No,74.0,Mkt&Fin,53.62,Placed,275000.0
212,213,M,67.00,Others,67.00,Others,Commerce,73.00,Comm&Mgmt,Yes,59.0,Mkt&Fin,69.72,Placed,295000.0
213,214,F,74.00,Others,66.00,Others,Commerce,58.00,Comm&Mgmt,No,70.0,Mkt&HR,60.23,Placed,204000.0


In [7]:
df.head()

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,1,M,67.00,Others,91.00,Others,Commerce,58.00,Sci&Tech,No,55.0,Mkt&HR,58.80,Placed,270000.0
1,2,M,79.33,Central,78.33,Others,Science,77.48,Sci&Tech,Yes,86.5,Mkt&Fin,66.28,Placed,200000.0
2,3,M,65.00,Central,68.00,Central,Arts,64.00,Comm&Mgmt,No,75.0,Mkt&Fin,57.80,Placed,250000.0
3,4,M,56.00,Central,52.00,Central,Science,52.00,Sci&Tech,No,66.0,Mkt&HR,59.43,Not Placed,NaN
4,5,M,85.80,Central,73.60,Central,Commerce,73.30,Comm&Mgmt,No,96.8,Mkt&Fin,55.50,Placed,425000.0


In [9]:
df.tail()

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
210,211,M,80.6,Others,82.0,Others,Commerce,77.6,Comm&Mgmt,No,91.0,Mkt&Fin,74.49,Placed,400000.0
211,212,M,58.0,Others,60.0,Others,Science,72.0,Sci&Tech,No,74.0,Mkt&Fin,53.62,Placed,275000.0
212,213,M,67.0,Others,67.0,Others,Commerce,73.0,Comm&Mgmt,Yes,59.0,Mkt&Fin,69.72,Placed,295000.0
213,214,F,74.0,Others,66.0,Others,Commerce,58.0,Comm&Mgmt,No,70.0,Mkt&HR,60.23,Placed,204000.0
214,215,M,62.0,Central,58.0,Others,Science,53.0,Comm&Mgmt,No,89.0,Mkt&HR,60.22,Not Placed,NaN


In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 215 entries, 0 to 214
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   sl_no           215 non-null    int64  
 1   gender          215 non-null    object 
 2   ssc_p           215 non-null    float64
 3   ssc_b           215 non-null    object 
 4   hsc_p           215 non-null    float64
 5   hsc_b           215 non-null    object 
 6   hsc_s           215 non-null    object 
 7   degree_p        215 non-null    float64
 8   degree_t        215 non-null    object 
 9   workex          215 non-null    object 
 10  etest_p         215 non-null    float64
 11  specialisation  215 non-null    object 
 12  mba_p           215 non-null    float64
 13  status          215 non-null    object 
 14  salary          148 non-null    float64
dtypes: float64(6), int64(1), object(8)
memory usage: 25.3+ KB


In [25]:
df.columns

Index(['sl_no', 'gender', 'ssc_p', 'ssc_b', 'hsc_p', 'hsc_b', 'hsc_s',
       'degree_p', 'degree_t', 'workex', 'etest_p', 'specialisation', 'mba_p',
       'status', 'salary'],
      dtype='object')

In [13]:
df.shape

(215, 15)

In [17]:
df.describe()

,sl_no,ssc_p,hsc_p,degree_p,etest_p,mba_p,salary
count,215.000000,215.000000,215.000000,215.000000,215.000000,215.000000,148.000000
mean,108.000000,67.303395,66.333163,66.370186,72.100558,62.278186,288655.405405
std,62.209324,10.827205,10.897509,7.358743,13.275956,5.833385,93457.452420
min,1.000000,40.890000,37.000000,50.000000,50.000000,51.210000,200000.000000
25%,54.500000,60.600000,60.900000,61.000000,60.000000,57.945000,240000.000000
50%,108.000000,67.000000,65.000000,66.000000,71.000000,62.000000,265000.000000
75%,161.500000,75.700000,73.000000,72.000000,83.500000,66.255000,300000.000000
max,215.000000,89.400000,97.700000,91.000000,98.000000,77.890000,940000.000000


In [19]:
df.duplicated()

0      False
1      False
2      False
3      False
4      False
       ...  
210    False
211    False
212    False
213    False
214    False
Length: 215, dtype: bool

In [23]:
df.duplicated().sum()

0

In [ ]:
# Maybe dupilcates are present
# Remove duplicates
# df=df.drop_duplicates()



In [27]:
df.isnull()

,sl_no,gender,ssc_p,ssc_b,hsc_p,hsc_b,hsc_s,degree_p,degree_t,workex,etest_p,specialisation,mba_p,status,salary
0,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False,False,False,False,True
4,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
210,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
211,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
212,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False
213,False,False,False,False,False,False,False,False,False,False,False,False,False,False,False


In [29]:
df.isnull().sum()

sl_no              0
gender             0
ssc_p              0
ssc_b              0
hsc_p              0
hsc_b              0
hsc_s              0
degree_p           0
degree_t           0
workex             0
etest_p            0
specialisation     0
mba_p              0
status             0
salary            67
dtype: int64

In [33]:
# Find the distribution
# one column
print(df['salary'].skew())  """For numeric columns only output - 0 --> normal distribution, otherwise it is considered as Skew distribution
                              if the distribution is skew use median to fill null values """

3.5697471998711054


In [39]:
median_salary = df["salary"].median()

print(median_salary)

265000.0


In [41]:
df["salary"] = df["salary"].fillna(median_salary)

In [43]:
df.isnull().sum()

sl_no             0
gender            0
ssc_p             0
ssc_b             0
hsc_p             0
hsc_b             0
hsc_s             0
degree_p          0
degree_t          0
workex            0
etest_p           0
specialisation    0
mba_p             0
status            0
salary            0
dtype: int64

In [51]:
columns = df.select_dtypes(include=['int64', 'float64']).columns
columns

Index(['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary'], dtype='object')

In [53]:
# find Outlier --- only for numerical columns 
columns=['sl_no', 'ssc_p', 'hsc_p', 'degree_p', 'etest_p', 'mba_p', 'salary']

In [55]:
# Find Outliers
for col in columns:
    q1=df[col].quantile(0.25)  # Quantile - first 25 % data mean
    q3=df[col].quantile(0.75) # Quantile - first 75% data mean
    iqr=q3-q1 # inner quantile
    lower_bound=q1 - 1.5 * iqr
    upper_bound=q3 + 1.5 * iqr
    Outliers=df[(df[col]>upper_bound) | (df[col]<lower_bound)]
    print(len(Outliers))
    

0
0
8
1
0
0
33


In [87]:
# Handle Outliers with clip function
for col in columns:

    q1 = df[col].quantile(0.25)
    q3 = df[col].quantile(0.75)

    iqr = q3 - q1

    lower_bound = q1 - 1.5 * iqr
    upper_bound = q3 + 1.5 * iqr

    df[col] = df[col].clip(lower=lower_bound,
                           upper=upper_bound)


In [61]:
# Recheck the ouliers
for col in columns:
    q1=df[col].quantile(0.25)  # Quantile - first 25 % data mean
    q3=df[col].quantile(0.75) # Quantile - first 75% data mean
    iqr=q3-q1 # inner quantile
    lower_bound=q1 - 1.5 * iqr
    upper_bound=q3 + 1.5 * iqr
    Outliers=df[(df[col]>upper_bound) | (df[col]<lower_bound)]
    print(len(Outliers))
    

0
0
0
0
0
0
0


                                                        " Data Analysis Using Pandas "

In [63]:
# What percentage of students got placed?
df["status"].value_counts(normalize=True) * 100

status
Placed        68.837209
Not Placed    31.162791
Name: proportion, dtype: float64

In [67]:
# 2. Gender-wise Placement
pd.crosstab(df["gender"], df["status"])


status,Not Placed,Placed
gender,,
F,28,48
M,39,100


In [73]:
print("Average Salary : ",df[df["status"]=="Placed"]["salary"].mean())

Average Salary :  269778.7162162162


In [75]:
print("Highest Salary :",df["salary"].max())

Highest Salary : 331250.0


In [89]:
#Specialization vs Placement 
pd.crosstab(df["specialisation"], df["status"])

status,Not Placed,Placed
specialisation,,
Mkt&Fin,25,95
Mkt&HR,42,53


In [91]:
# Specialization vs Salary
df.groupby("specialisation")["salary"].mean()

specialisation
Mkt&Fin    271664.583333
Mkt&HR     264026.315789
Name: salary, dtype: float64

In [77]:
# Work Experience Impact
pd.crosstab(df["workex"], df["status"])

status,Not Placed,Placed
workex,,
No,57,84
Yes,10,64


6. Degree Stream Analysis
Which degree stream gets placed most?

In [81]:
pd.crosstab(df["degree_t"],df["status"])

status,Not Placed,Placed
degree_t,,
Comm&Mgmt,43,102
Others,6,5
Sci&Tech,18,41


In [85]:
print("Key Findings")
print("- Placement Rate:", round(df["status"].value_counts(normalize=True)["Placed"]*100,2),"%")
print("- Average Salary:", round(df[df["status"]=="Placed"]["salary"].mean(),2))
print("- Highest Salary:", df["salary"].max())

Key Findings
- Placement Rate: 68.84 %
- Average Salary: 269778.72
- Highest Salary: 331250.0


Findings From this Project

1. Approximately 69% of students secured placements, indicating a strong overall placement rate.

2. Students with prior work experience showed significantly better placement outcomes compared to students without work experience.

3. Marketing & Finance specialization recorded both higher placement counts and slightly higher average salaries than Marketing & HR specialization.

4. Commerce & Management students achieved the highest number of placements among all degree streams.

5. The average salary offered to placed students was ₹269,779, while the highest salary reached ₹331,250.

6. Academic performance appeared to positively influence placement success, with placed students generally achieving higher academic scores.

7. Placement outcomes were influenced by multiple factors including work experience, specialization, degree stream, and academic performance.